discovery_engine_core/providers/src/client.rs

```diff
@@ -219,10 +200,41 @@ impl Client {
     }
 
     /// Run a query for fetching `Article`s from Newscatcher.
-    pub async fn query_articles(&self, query: &impl Query) -> Result<Vec<Article>, Error> {
-        self.query_newscatcher(query)
+    pub async fn query_articles(&self, query: &impl Query) -> Result<Vec<GenericArticle>, Error> {
+        let articles = self
+            .query_newscatcher(query)
             .await
-            .map(|news| news.articles)
+            .map(|news| news.articles)?;
+
+        // If we don't receive any articles from the news source, we don't treat this
+        // as an error...
+        if articles.is_empty() {
+            return Ok(vec![]);
+        }
+
+        let generic_articles: Vec<GenericArticle> = articles
+            .into_iter()
+            .filter_map(|article| {
+                GenericArticle::try_from(article.clone())
+                    .map_err(|e| {
+                        trace!(
+                            "Malformed article could not be convert ({:?}): {:?}",
+                            e,
+                            article
+                        )
+                    })
+                    .ok()
```

Comment:
🔧 Logging the error should be enough 

```suggestion
                GenericArticle::try_from(article)
                    .map_err(|e| {
                        error!("Malformed article could not be converted: {}", e);
                    })
                    .ok()
```